airflow/providers/google/cloud/hooks/bigquery.py

```diff
@@ -160,6 +161,27 @@ def get_client(self, project_id: Optional[str] = None, location: Optional[str] =
             credentials=self._get_credentials(),
         )
 
+    def get_uri(self) -> str:
+        """Override DbApiHook get_uri method for get_sqlalchemy_engine()"""
+        return "bigquery://"
+
+    def get_sqlalchemy_engine(self, engine_kwargs=None):
+        """
+        Get an sqlalchemy_engine object.
+
+        :param engine_kwargs: Kwargs used in :func:`~sqlalchemy.create_engine`.
+        :return: the created engine.
+        """
+        connection = self.get_connection(self.gcp_conn_id)
+        extras = connection.extra_dejson
+        credentials_path = extras.get("extra__google_cloud_platform__key_path", None)
+        if credentials_path is None:
+            raise AirflowException(
+                "For now, we only support instantiating SQLAlchemy engine by"
```

Comment:
Should we also support [ADC](https://cloud.google.com/docs/authentication/production#automatically)? This is very often used in a production environment, and I think it won't be hard to add.

Reply:
Adding `credential_info` support shouldn't be difficult either. Have you encountered any problems with it?

Reply:
on it~

Reply:
FYI https://github.com/apache/airflow/pull/19508#discussion_r798499356